src/accountsdb/download.zig

```diff
@@ -538,10 +544,171 @@ pub fn downloadFile(
     }
 }
 
-const LegacyContactInfo = sig.gossip.data.LegacyContactInfo;
-const SignedGossipData = sig.gossip.data.SignedGossipData;
+pub fn getOrDownloadAndUnpackSnapshot(
+    allocator: std.mem.Allocator,
+    logger_: Logger,
+    validator_dir: std.fs.Dir,
+    options: struct {
+        /// gossip service is not needed when loading from an existing snapshot.
+        /// but when we need to download a new snapshot (force_new_snapshot_download flag),
+        /// we need the gossip service.
+        gossip_service: ?*GossipService = null,
+        force_new_snapshot_download: bool = false,
+        force_unpack_snapshot: bool = false,
+        num_threads_snapshot_unpack: u16 = 0,
```

Comment:
maybe if I/O operations are really slow then you could throw more threads at it and it could work but in most cases, more than 2x cpu_count wont make it faster